Remotion/Core/UnitTests/Context/SafeContextTest.Thread.cs

```diff
@@ -0,0 +1,149 @@
+﻿using System;
+using System.Runtime.ExceptionServices;
+using System.Threading;
+using Moq;
+using NUnit.Framework;
+using Remotion.Context;
+
+namespace Remotion.UnitTests.Context
+{
+  public partial class SafeContextTest
+  {
+    [TestFixture]
+    public class ThreadTest
```

Comment:
different design for the tests -> not nested types.

most conform choice would be a folder SafeContextTests with individual partial files as separate text fixtures. Please update

Reply:
What is the problem with nested types in this scenario? We use nested types in the implementation and the tests should mirror that structure in my opinion.

Reply:
It's an example where we want Testcase Class per Feature (http://xunitpatterns.com/Testcase%20Class%20per%20Feature.html) because each nested type of the SUT represents a feature. We never use nested test fixtures and it would introduce a new concept and thus new complexity and thus reduce maintainability of the tests. 

The idea when structuring the tests is to make them a) findable and b) allow shared setups, test data, etc. By using a nested test fixture, in particular for a case where there's tests on both the outer type and the nested type, makes the tests much harder to read and understand their interdependencies and possible sideeffects. That's why separate test fixtures in a shared folder/namespace is the pattern used for this case.